<launch> 
	<arg name="cam_id"            default="0"                         /> 
	<arg name="ar_node_name"      default="individualMarkersNoKinect" />
	<arg name="bundle_files"      default=""                          />
    <arg name="marker_size"       default="50.0"                      />
    <arg name="calibration_file"  default="$(find alvar_tag_tracking)/share/calibration/webcam/webcam.yaml"/>

	<!-- Parameters --> 
	<arg name="cam_name"             default="usb_cam_$(arg cam_id)"                />
	<arg name="marker_size"          default="5.7"                                  />
	<arg name="max_new_marker_error" default="0.08"                                 />
	<arg name="max_track_error"      default="0.2"                                  />
	<arg name="pose_marker_topic"    default="/ar_pose_marker"                      />
	<arg name="cam_frame_id"         default="$(arg cam_name)"                      />
	<arg name="cam_image_topic"      default="/$(arg cam_name)/image_raw"           />
	<arg name="cam_info_topic"       default="/$(arg cam_name)/camera_info"         />

	<group ns="$(arg cam_name)">
		<remap from="/$(arg cam_name)/usb_cam/image_raw"    to="/$(arg cam_name)/image_raw"  />
		<remap from="/$(arg cam_name)/usb_cam/camera_info"  to="/$(arg cam_name)/camera_info"/>
		<remap from="/$(arg cam_name)/ar_pose_marker"       to="/ar_pose_marker"             />
		<remap from="/$(arg cam_name)/visualization_marker" to="/visualization_marker"       />

		<!-- Camera --> 
		<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
			<param name="video_device"    value="/dev/video$(arg video_id)"                            />
			<param name="camera_info_url" value="file://$(find projet)/share/usb_cam_$(arg cam_id).yml"/>
			<param name="camera_frame_id" value="$(arg cam_frame_id)"                                  />
			<param name="pixel_format"    value="yuyv"                                                 />
		</node>

		<!-- Marker Detector -->
		<node name="ar_track_alvar" pkg="ar_track_alvar" type="$(arg ar_node_name)" respawn="false" output="screen">
			<param name="marker_size"          type="double" value="$(arg marker_size)"                             />
			<param name="max_new_marker_error" type="double" value="$(arg max_new_marker_error)"                    />
			<param name="max_track_error"      type="double" value="$(arg max_track_error)"                         />
			<param name="output_frame"         type="string" value="$(arg cam_frame_id)"                            />
			<param name="bundle_files"         type="string" value="$(find projet)/bundles/xmls/$(arg bundle_files)"/>

			<remap from="/tf"          to="/tf0"                   />
			<remap from="camera_image" to="$(arg cam_image_topic)" />
			<remap from="camera_info"  to="$(arg cam_info_topic)"  />  
		</node>	
	</group>
</launch>
